kotlinx-coroutines-core/common/src/CancellableContinuation.kt

```diff
@@ -41,6 +41,8 @@ import kotlin.coroutines.intrinsics.*
  *    +-----------+
  * ```
  */
+@OptIn(ExperimentalSubclassOptIn::class)
```

Comment:
just in case, may be it could be easier to just `optIn` globally for `ExperimentalSubclassOptIn` via `compilerOptions`?

Reply:
It would be easier, but the way it is now, having to opt-in is helping us: in every place where there's an `ExperimentalSubclassOptIn` required, we explicitly decide whether to opt-in or to propagate.

Reply:
yeah, overall, I understand the difference and agree. But the case with `ExperimentalSubclassOptIn` is IMO a little more specific, as propagating it, will mean that on every usage of annotated class (not only inheriting) the warning will be issues about `ExperimentalSubclassOptIn` feature :)
So propagating this specific optIn doesn't make any sense to me. That's why I mentioned the `global` optIn

Reply:
Oh, no, in this case, by "propagating (to the inheritor)" I mean adding `SubclassOptInRequired`. Sometimes, it's okay to inherit from an inheritor, but not from the base class directly, and sometimes, inheriting an inheritor is also undesirable.